### PR TITLE
CLI - Project - Add Service, create directory and add starter pipelin…

### PR DESCRIPTION
**Pull request by mtarng**
…es.yaml file

initial PR for https://github.com/microsoft/bedrock/issues/577

### CODE REVIEW

File: src/commands/project/add-service.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import commander from "commander";
+import fs from "fs";
+import yaml from "js-yaml";
+import path from "path";
+import shell from "shelljs";
+import mkdirp from "mkdirp";
```
Comment (evanlouie) [inline src/commands/project/add-service.ts]: should keep on using shelljs instead. you can do `shell.mkdir('-p', 'your-dir')`

File: src/commands/project/add-service.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import { promisify } from "util";
+import { logger } from "../../logger";
+import { generateAzurePipelinesYaml } from "../../lib/fileutils";
+import { IBedrockFile, IMaintainersFile } from "../../types";
+
+// npm install @types/mkdirp
```
Comment (evanlouie) [inline src/commands/project/add-service.ts]: dont need

File: src/commands/project/add-service.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  logger.info(`Adding Service: ${serviceName}, to Project: ${rootProjectPath}`);
+  logger.info(
+    `MaintainerName: ${maintainerName}, MaintainerEmail: ${maintainerEmail}`
+  );
+
+  // Mkdir
+  const newServiceDir = path.join(rootProjectPath, serviceName);
+
+  mkdirp(`${newServiceDir}`, function(err) {
```
Comment (evanlouie) [inline src/commands/project/add-service.ts]: use `shell.mkdir('-p', newServiceDir)` instead

File: package.json [hunk 4]
```diff
@@ -47,6 +47,7 @@
   },
   "dependencies": {
     "cli-table": "^0.3.1",
+    "@types/mkdirp": "^0.5.2",
```
Comment (evanlouie) [inline package.json]: remove

Comment (evanlouie) [reply package.json]: Actually not sure how this is working because we dont have `mkdirp` as a dependency

**Review by evanlouie — CHANGES_REQUESTED** [review]

File: src/commands/project/add-service.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+/**
+ * Adds the init command to the commander command object
+ *
+ * @param command Commander command object to decorate
+ */
+export const addServiceCommandDecorator = (
+  command: commander.Command
+): void => {
+  command
+    .command("add-service <service-name>")
```
Comment (timfpark) [inline src/commands/project/add-service.ts]: For the `deployment` command, we separate nouns and verbs into two separate parts, eg.

`spk deployment get`

but here we are combining them into one eg.:

`spk add-service`

I would propose that we agree on one way of doing this and one common set of verbs (where possible) such that the cli feels like a coherent whole in the same way that the `kubectl` tool does despite having been developed by hundreds of engineers.

**Review by timfpark — CHANGES_REQUESTED** [review]
Let's agree on a common grammar for commands across the different teams - otherwise looks great to me.

File: src/commands/project/add-service.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+      } catch (err) {
+        logger.error(
+          `Error occurred adding service ${serviceName} to project ${projectPath}`
+        );
+        logger.error(err);
+      }
+    });
+};
+
+export const addService = async (
```
Comment (evanlouie) [inline src/commands/project/add-service.ts]: Add some JSDoc please 🙏 
We should standardize on having all export/public functions/classes on having some accompanying docs

**Review by evanlouie — CHANGES_REQUESTED** [review]

File: src/commands/project/add-service.ts (hunk 5 above)
Comment (mtarng) [reply src/commands/project/add-service.ts]: This command was initially under `spk project add-service`, but after speaking with @evanlouie we're moving service creation/management under it's own command; now it's `spk service create` etc.

**Review by evanlouie — APPROVED** [review]

**Review by bnookala — APPROVED** [review]

**Pull request merged**